app/connectors/ApiSubscriptionFieldsConnector.scala

```diff
@@ -130,14 +122,22 @@ abstract class AbstractSubscriptionFieldsConnector(implicit ec: ExecutionContext
     } recover recovery(DefinitionsByApiVersion.empty)
   }
 
-  def saveFieldValues(clientId: String, apiContext: String, apiVersion: String, fields: Fields)(
-      implicit hc: HeaderCarrier
-  ): Future[HttpResponse] = {
+  def saveFieldValues(clientId: String, apiContext: String, apiVersion: String, fields: Fields)
+                     (implicit hc: HeaderCarrier): Future[SaveSubscriptionFieldsResponse] = {
     val url = urlSubscriptionFieldValues(clientId, apiContext, apiVersion)
-    http.PUT[SubscriptionFieldsPutRequest, HttpResponse](
-      url,
-      SubscriptionFieldsPutRequest(clientId, apiContext, apiVersion, fields)
-    )
+
+    import CustomResponseHandlers.permissiveBadRequestResponseHandler
+
+    http.PUT[SubscriptionFieldsPutRequest, HttpResponse](url, SubscriptionFieldsPutRequest(clientId, apiContext, apiVersion, fields)).map { response =>
+      response.status match {
+        case BAD_REQUEST =>
+          SaveSubscriptionFieldsFailureResponse(
+            Json.fromJson[Map[String, String]](
```

Comment:
should this not use validate rather than fromJson.  Validate gives an either rather than throwing if it fails to convert.

Reply:
Depends on what you want to happen on an invalid payload on a 400.

It's not really recoverable so I don't know.

Reply:
Also, what validate do you mean?

Reply:
Do you mean .validate on the result of calling Json.parse?
If so we could do that I suppose. Still not sure what to do when the response body isn't correct though.

Reply:
I do.  Ok.  I guess so long as it ends up as a 5xx somewhere..?